### DISCUSSION TITLE
Common root folder for different languages

### DISCUSSION THREAD

**Discussion by MichaelStather** [facebook/docusaurus#7587] [Q&A]
Hi,

I`m creating a Docusaurus based documentation system. The actual .mdx files are generated automatically from the source code of the product.
Now I have a question about **i18n**:

As seen in the Docusaurus documentation the root folder for content is:

**/docs** for the default language
**/i18n/de/docusaurus-plugin-content-docs/current** for e.g. the german version

This is not ideal for our auto-generating workflow, we would like to have the content root folders like this:

**/docs/en**
**/docs/de**

or

**/docs-en**
**/docs-de**

So that all content, independent of language, is inside a **commmon root folder**.

Is this possible?

Thanks for your help ;)

**Comment by slorber** [comment]
@MichaelStather we don't support this for now, the i18n paths are not very flexible and you have to conform to our convention

@Josh-Cena recently implemented the ability to configure paths per locale: https://github.com/facebook/docusaurus/pull/7386

This means you could have something like:

```
/docs
/de/docusaurus-plugin-content-docs/current
```

We don't have yet:
- the ability to not have `current` in the fs path, for unversioned sites
- the ability to configure the localized path of a plugin
- the ability to make the master language look like any other language (ie have a `/en` folder instead of `docs`)

But we definitively want more flexibility in this i18n system in the long term
